workshops/boost-research-reproducibility-binder/workshop-presentations/zero-to-binder.md

```diff
@@ -0,0 +1,254 @@
+# From Zero to Binder!
+
+Sarah Gibson, _The Alan Turing Institute_
+
+[**The Turing Way**](https://github.com/alan-turing-institute/the-turing-way) - making reproducible Data Science "too easy not to do"
+
+Based on Tim Head's _Zero-to-Binder_ workshops which can be found here: http://bit.ly/zero-to-binder and http://bit.ly/zero-to-binder-rise
+
+To follow these instructions on your own machine, follow this link: **http://bit.ly/SG-zero-to-binder**
+
+## Running Code is more complicated than Displaying Code
+
+GitHub is a great service for sharing code, but the contents are **static**.
+
+How could you _run_ a GitHub repository **without installing complicated requirements**?
+Or even **in your browser**?
+
+To run code, you need:
+* Hardware on which to run the code
+* Software, including:
+  * The code itself
+  * The programming language (e.g. Python, R)
+  * Relevant packages (e.g. pandas, matplotlib, tidyverse, ggplot)
+
+## What Binder Provides
+
+[Binder](https://mybinder.org) is a service that provides your code and the hardware and software to execute it.
+
+You can create a link to a **live, interactive** version of your code!
+
+* An example binder link:
+
+  > **https://mybinder.org/v2/gh/trekhleb/homemade-machine-learning/master?filepath=notebooks%2Fanomaly_detection%2Fanomaly_detection_gaussian_demo.ipynb**
+
+  From this repo: **https://github.com/trekhleb/homemade-machine-learning**
+
+  * Notice that the Binder link has a similar structure to the GitHub repo link
+  * The "filepath" argument opens a specific notebook in the repo
+
+## 1. Creating a repo to Binderize
+
+**TO DO:** :vertical_traffic_light:
+
+1) Create a new repo on GitHub called "my-first-binder".
+   * Don't forget to initialise with a README!
+2) Create a file called `hello.py` via the web interface with `print("Hello from Binder!")` on the first line and commit to master
+   * Bad practice :see_no_evil: but we're not collaborating right now!
+
+## 2. Launch your first repo!
+
+**TO DO:** :vertical_traffic_light:
+
+1) Go to **https://mybinder.org**
+2) Type the URL of your repo into the "GitHub repo or URL" box.
+
+   Should look like this:
+   > **https://github.com/your-username/my-first-binder**
+
+3) As you type, the webpage generates a link in the "Copy the URL below..." box
+
+   Should look like this:
+   > **https://mybinder.org/v2/gh/your-username/my-first-binder/master**
+
+4) Copy it, open a new browser tab and visit that URL.
+   * You will see a "spinner" as Binder launches the repo.
+
+If everything ran smoothly, you'll see a Jupyter Notebook interface.
+
+#### What's happening in the background?
+
+While you wait, BinderHub (the backend of Binder) is:
+* Fetching your repo from GitHub
+* Analysing the contents
+* Creating a Docker file
```

Comment:
"Creating a Docker file based on your repository" might be more helpful for anyone reading these notes after your presentation.

Reply:
Clarified in 
[`0a9b8eb`](/alan-turing-institute/the-turing-way/pull/235/commits/0a9b8ebcacb089f40978d9ccf4a8920c9b0486c7)